### PR TITLE
Possibility to specify an update check interval for the web worker.

### PR DESCRIPTION
**Pull request by oderwat**
With this change, it is possible to specify an interval at which the server is checked for updates. Set this to one hour or just some seconds (while development) to have the web worker update itself and trigger the `OnAppUpdate` handler without manually reloading the page or the need for the user to browse to another URL. This combined with a UX dialog that tells the user that an update is pending should solve the problem with long-running apps that only use internal routes and API calls without ever loading another page from the server itself.

### CODE REVIEW

File: pkg/app/http.go [hunk 1]
```diff
@@ -346,13 +349,15 @@ func (h *Handler) makeAppJS() []byte {
...
+			WorkerJS       string
+			UpdateInterval string
 		}{
-			Env:      btos(env),
-			Wasm:     h.Resources.AppWASM(),
-			WorkerJS: h.resolvePackagePath("/app-worker.js"),
+			Env:            btos(env),
+			Wasm:           h.Resources.AppWASM(),
+			WorkerJS:       h.resolvePackagePath("/app-worker.js"),
+			UpdateInterval: strconv.Itoa(h.UpdateInterval),
```
Comment (maxence-charriere) [inline pkg/app/http.go:360]: With the auto-update interval set as a duration here would have to be changed to:
```go
AutoUpdateInterval: AutoUpdateInterval.Milliseconds(),
```

File: pkg/app/http.go [hunk 2]
```diff
@@ -346,13 +349,15 @@ func (h *Handler) makeAppJS() []byte {
 	if err := template.
 		Must(template.New("app.js").Parse(appJS)).
 		Execute(&b, struct {
-			Env      string
-			Wasm     string
-			WorkerJS string
+			Env            string
+			Wasm           string
+			WorkerJS       string
+			UpdateInterval string
```
Comment (maxence-charriere) [inline pkg/app/http.go:355]: From https://developer.mozilla.org/en-US/docs/Web/API/setInterval, a delay is a number in milliseconds.

Here can use:
```go
AutoUpdateInterval int64
```

File: pkg/app/gen/app.js [hunk 3]
```diff
@@ -3,6 +3,8 @@
 // -----------------------------------------------------------------------------
 var goappOnUpdate = function () { };
 
+const updateInterval = {{.UpdateInterval}};
```
Comment (maxence-charriere) [inline pkg/app/gen/app.js:6]: Would be good to have this named `autoUpdateInterval`:

```js
const autoUpdateInterval = {{.AutoUpdateInterval}};
```

**Review by maxence-charriere — CHANGES_REQUESTED** [review]
This seems to be a pretty useful change. Thanks for writing this PR.

Just a few things that could be polished and I'll gladly merge this.

**Comment by oderwat** [comment]
I made the change you proposed. Very nice!

**Review by maxence-charriere — APPROVED** [review]

**Pull request merged**